### PR TITLE
Cleaning up FilterAdapter and its usages

### PR DESCRIPTION
**Pull request by florina-muntenescu**
Finally the data layer doesn't have any more references to the UI layer! 🎉

### CODE REVIEW

File: app/src/main/java/io/plaidapp/ui/HomeActivity.java [hunk 1]
```diff
@@ -119,16 +120,17 @@
...
     private boolean monitoringConnectivity = false;
+    private FilterAdapter filtersAdapter;
 
     // data
     @Inject
     DataManager dataManager;
     @Inject
     FeedAdapter adapter;
     @Inject
-    FilterAdapter filtersAdapter;
```
Comment (ataulm) [inline app/src/main/java/io/plaidapp/ui/HomeActivity.java]: 🙌

File: core/src/main/java/io/plaidapp/core/data/prefs/SourcesRepository.kt [hunk 2]
```diff
@@ -27,6 +28,12 @@ class SourcesRepository(
     private val dataSource: SourcesLocalDataSource
 ) {
 
+    private val callbacks = mutableListOf<FiltersChangedCallback>()
+
+    fun registerFilterChangedCallback(callback: FiltersChangedCallback) {
```
Comment (ataulm) [inline core/src/main/java/io/plaidapp/core/data/prefs/SourcesRepository.kt]: is this pattern expected to stay here long term?  I would expect eventually that `getSources()` would return `LiveData<List<Source>>`/`Observable<List<Source>>`.

File: core/src/main/java/io/plaidapp/core/ui/filter/FilterAdapter.kt [hunk 3]
```diff
@@ -35,18 +35,14 @@ class FilterAdapter(
     private val sourcesRepository: SourcesRepository
 ) : RecyclerView.Adapter<FilterViewHolder>(), FilterSwipeDismissListener {
 
-    private var callbacks: MutableList<FiltersChangedCallback> = ArrayList()
-
     val enabledSourcesCount: Int
         get() = filters.count(Source::active)
 
-    private val _filters: MutableList<Source>
-    val filters: List<Source>
```
Comment (ataulm) [inline core/src/main/java/io/plaidapp/core/ui/filter/FilterAdapter.kt]: Was this unused already? I can't see usages of `.filters` in this PR.

File: core/src/main/java/io/plaidapp/core/data/prefs/SourcesRepository.kt (hunk 2 above)
Comment (florina-muntenescu) [reply core/src/main/java/io/plaidapp/core/data/prefs/SourcesRepository.kt]: yeah, we're definitely going to change it to something better than callbacks. we haven't decided on what for now.

File: core/src/main/java/io/plaidapp/core/ui/filter/FilterAdapter.kt (hunk 3 above)
Comment (florina-muntenescu) [reply core/src/main/java/io/plaidapp/core/ui/filter/FilterAdapter.kt]: it was used from `DataManger`, for example, which is still a java file, so it was `.getFilters`

Comment (ataulm) [reply core/src/main/java/io/plaidapp/core/ui/filter/FilterAdapter.kt]: ♂️oh yea

**Pull request merged**